Monolog/Processor/GitProcessor.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Troopers\MetricsBundle\Monolog\Processor;
+
+class GitProcessor
+{
+    private $kernelRootDir;
+    private static $cache = false;
+
+    /**
+     * GitProcessor constructor.
+     *
+     * @param $kernelRootDir
+     */
+    public function __construct($kernelRootDir)
+    {
+        $this->kernelRootDir = $kernelRootDir;
+    }
+
+    public function __invoke(array $record)
+    {
+        $record['extra']['@git'] = $this->getRevision();
+
+        return $record;
+    }
+
+    private function getRevision()
+    {
+        if (self::$cache !== false) {
+            return self::$cache;
+        }
+
+        //Get revision number from .git
+        if ($gitRev = shell_exec('git rev-parse HEAD')) {
```

Comment:
I'd surround the `git rev-parse HEAD` to be sure there is no problem running this command and its output is well a hash as expected